packages/ocl/src/Util.js

```diff
@@ -0,0 +1,106 @@
+import { composeNextState } from '@openfn/language-common';
+
+export function buildUrl(configuration, options) {
+  const {
+    ownerId,
+    ownerType,
+    repository,
+    repositoryId,
+    version,
+    expansions,
+    expansionId,
+    content,
+    query,
+    ...rest
+  } = options;
+
+  const { baseUrl } = configuration;
+
+  const urlParts = [
+    baseUrl,
+    ownerType,
+    ownerId,
+    repository,
+    repositoryId,
+    version,
+    expansions,
+    expansionId,
+    content,
+  ].filter(urlPart => urlPart);
+  const url = urlParts.join('/');
+
+  return { url, query: { ...query, ...rest } };
+}
+
+export function isObjectEmpty(objectName) {
+  return (
+    objectName &&
+    Object.keys(objectName).length === 0 &&
+    objectName.constructor === Object
+  );
+}
+
+export function handleResponse(response, state, callback) {
+  const nextState = {
+    ...composeNextState(state, response),
+    response,
+  };
+  if (callback) return callback(nextState);
+  return nextState;
+}
+
+export function handleError(error) {
+  throw error;
```

Comment:
what's the purpose of `.catch(e => throw e)` ?

Reply:
The idea was `handleError()` sometimes have `error.response` object. If that happens you might want to do something like this 
```
 if (error.response) {
    const { method, path } = error.response.request;
    const { status } = error.response;
    if (Object.keys(error.response.data).length === 0) {
      throw new Error(
        `Server responded with:  \n${JSON.stringify(error.response, null, 2)}`
      );
    }
 ```
 Basically any custom error handling should happen in `handleError()` function. Since we don't have that use case yet, i am replacing this `handleError()` function with a simple `.catch(e => {throw e})`